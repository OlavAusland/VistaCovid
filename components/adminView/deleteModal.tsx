import { Modal, Pressable, Text, View } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";

type DeleteRoomModal={
    modalVisible: boolean;
    handleRequestClose: Function;
    deleteRoomAndClose: Function;
}

export const DeleteRoomModal = (props: DeleteRoomModal) => {
    
    return(
        <Modal
        animationType="slide"
        statusBarTranslucent={true}
        transparent={true}
        visible={props.modalVisible}
        onRequestClose={()=>props.handleRequestClose()}>

                <View style={adminStyle.centeredView}>
                    <View style={adminStyle.modalView}>
                        <Text style={adminStyle.modalText}>Are you sure you want to delete this room?</Text>
                        
                        <Pressable
                        style={[adminStyle.buttonClose, adminStyle.buttonClose]}
                        onPress={()=> props.deleteRoomAndClose()}
                        >
                            <Text style={adminStyle.textStyle}>Yes, delete room</Text>
                        </Pressable>

                        <Pressable
                        style={[adminStyle.buttonClose, adminStyle.buttonClose]}
                        onPress={()=> props.handleRequestClose()}
                        >
                            <Text style={adminStyle.textStyle}>No, close modal</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
    );
}

