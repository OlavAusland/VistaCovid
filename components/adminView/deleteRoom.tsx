import { ScrollView } from "react-native";


export function DeleteRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}