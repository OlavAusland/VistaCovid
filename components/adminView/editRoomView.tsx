import { ScrollView } from "react-native";


export function EditRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}