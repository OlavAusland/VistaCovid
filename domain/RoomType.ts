import { Roles } from "./UserType"

export type GraphData = {
    time: number,
    value: number
}

export type NoteData = {
    role: string,
    note: string
}

export type Room = {
    patientId: string,
    lastUpdated: string | undefined,
    heartRate: Array<GraphData> 
    bloodPressure: Array<GraphData> 
    oxygenLevel: Array<GraphData> 
    notes: Array<NoteData> | undefined,
    id: string
}

export type csvrooms = {
    data: Array<helthData>
}

type helthData = {
    bloodPressure: Array<GraphData> 
    heartRate: Array<GraphData>
    oxygenLevel: Array<GraphData>
}


