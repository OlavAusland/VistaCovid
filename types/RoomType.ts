export type GraphData = {
    time: string,
    value: number
}

export type Room = {
    patientId: string,
    roomNumber: string,
    lastUpdated: string | undefined,
    heartRate: Array<GraphData> | undefined,
    breathingRate: Array<GraphData> | undefined,
    bloodPressure: Array<GraphData> | undefined,
    oxygenLevel: Array<GraphData> | undefined,
    notes: Array<GraphData> | undefined
}