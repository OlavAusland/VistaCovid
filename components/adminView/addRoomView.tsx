import { useState } from "react";
import React, { ScrollView, TextInput, View, Text, Button, SafeAreaView, Alert } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";
import { Room } from "../../domain/RoomType";
import { addRoom, getRooms } from "../../api/firebaseAPI";
import { NoteData } from "../../domain/RoomType";
import { Roles } from "../../domain/UserType";

export function AddRoom()
{

    const [roomNumber, setRoomNumber] = useState("")

 
    const newroom: Room = {
        id: roomNumber,
        patientId: "",
        roomNumber: roomNumber,
        lastUpdated: Date.now().toString(),
        heartRate: [],
        bloodPressure: [],
        oxygenLevel: [],
        notes: []
     }


     const SuccessfullAlert = () =>
     Alert.alert(
       "Successfully added room with room number: "+{roomNumber},
       "",
       [
         { text: "OK", onPress: () => console.log("OK Pressed", roomNumber) }
       ]
     );

     const UnsuccessfullAlert = () =>
     Alert.alert(
       "Could not add room with room number: ",
       "Room might already exist",
       [
         { text: "OK", onPress: () => console.log("OK Pressed") }
       ]
     );

     const AddRoom = () =>{
         addRoom(newroom).then(() => {SuccessfullAlert()}).catch(() => {UnsuccessfullAlert()})
     }

    return(
        <ScrollView style={adminStyle.addRoomContainer}>
            <SafeAreaView>
                <View style={adminStyle.addRoomHeader}>
                    <Text style={adminStyle.headertext}>Add Room</Text>
                </View>
                <View>
                    <TextInput
                    style={adminStyle.addRoomInput}
                    onChangeText={setRoomNumber}
                    placeholder="room number"/>
                    <View style={adminStyle.addRoomButton}>
                        <Button title='Add Room' onPress={()=>{AddRoom()}}/>
                    </View>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}