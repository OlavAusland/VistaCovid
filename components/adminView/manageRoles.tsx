import { useEffect, useState } from "react";
import React, { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable, FlatList, TouchableOpacity, StyleSheet } from "react-native";
import { SearchBar } from 'react-native-elements';

// navigation
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParameters } from '../../domain/NavigationTypes';
import { adminStyle } from "../../styles/AdminStyles";
import { Room } from "../../domain/RoomType";
import { User } from "../../domain/UserType";
import { deletePatient, getUsers, deleteUserById } from "../../api/firebaseAPI";
import { roomStyle } from "../../styles/RoomStyles";
import { isPropertySignature } from "typescript";
import { ErrorType } from "../../domain/Errortype";
import { Errormodal } from "../ErrorModal";
import Icon from "react-native-vector-icons/Feather";

export function ManageRoles(){

    const navigation = useNavigation<NativeStackNavigationProp<StackParameters>>();

    const [searchWord, setSearchWord] = useState("");
    const [users, setUsers] = useState<User[]>([])
    const [search, setSearch] = useState<string>('');
    const [error, setError] = useState<ErrorType>({errorObject:undefined, errormodalVisible:false});

    const handleRequestClose = () => {
        setError((prev) =>({...prev,errorObject:undefined, errormodalVisible:false}));
    }

    useEffect(() => {
        const getUserData = async () =>{
            await getUsers().then((res) => {
                setUsers(res);
            }).catch((err) => { setError((prev) =>({...prev, errorObject:err, errormodalVisible:true})); });
        };
        getUserData();
    }, []);

    if(error.errormodalVisible){
        return (
            <Errormodal error={error} handleRequestClose={handleRequestClose} />
        )
    }

    return(
        <SafeAreaView style={{flex:1}}>
            <View style={[styles.header, styles.shadow]}>
                <TextInput
                    placeholder="room number"
                    onChangeText={(text) => {setSearch(text)}}
                    style={styles.search}/>
            </View>
            <ScrollView contentContainerStyle={{flex:1, alignItems:'center'}}>
                {users.map((user) => {
                    return(
                        <View  key={'users' + user.id} style={styles.cardContainer}>
                            <View style={styles.card}>
                                <Icon style={{alignSelf:'center'}} name='clipboard' size={80}/>
                                <Text style={{flex:1}}>Patient: {user.id}</Text>
                            </View>
                            <TouchableOpacity onPress={() =>{deleteUserById(user.id)}} style={styles.delete}>
                                <Icon name='trash-2' color={'white'} size={40}/>
                            </TouchableOpacity>
                        </View>
                    );
                })}
            </ScrollView>
        </SafeAreaView>
    );
}

export const styles = StyleSheet.create({
    header:{
        backgroundColor:'white',
        paddingTop:40,
        paddingBottom:10
    },
    delete: {
        backgroundColor:'red', 
        flexBasis:'15%', 
        height:100, 
        justifyContent:'center', 
        alignItems:'center', 
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    cardContainer:{
        width:'90%',
        marginTop:25,
        flexDirection:'row', 
        justifyContent:'space-between', 
        alignItems:'center', 
    },
    shadow:{
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3,},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    card: {
        flexDirection:'row', 
        backgroundColor:'#79CAED', 
        flexBasis:'84%', 
        height:100, 
        justifyContent:'center', 
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3,},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    add:{
        marginTop:25,
        backgroundColor:'white',
        width:'90%',
        height:75,
        justifyContent:'center',
        alignItems:'center',
        borderRadius:10
    },
    search:{
        height: 40,
        margin: 12,
        borderWidth: 1,
        borderRadius:10,
        padding: 10,
    }
})