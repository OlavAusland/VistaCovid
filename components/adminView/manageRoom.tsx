import { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable } from "react-native";
import { useState } from "react";
import { deleteRoom } from "../../api/firebaseAPI";
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParameters } from "../../domain/NavigationTypes";
import { adminStyle } from "../../styles/AdminStyles";
import  handleDelete from "./deleteModal"

export function ManageRoom()
{
    const navigation = useNavigation<NativeStackNavigationProp<StackParameters>>();

    const [roomNumber, setRoomNumber] = useState('')
    const [delroomNumber, setDelRoomNumber] = useState('')



    return(
        <ScrollView>
            <SafeAreaView>
                <View style={adminStyle.editRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setRoomNumber}/>
                    <View style={adminStyle.manageRoomButtons}>
                        <Button title="Edit room" onPress={()=>navigation.navigate("EditRoom")}/>
                    </View>
                </View>

                <View style={adminStyle.deleteRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setDelRoomNumber}/>
                    <Button title="DeleteRoom" onPress={()=>handleDelete}/>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}