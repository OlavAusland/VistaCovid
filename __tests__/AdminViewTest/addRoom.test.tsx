import { render } from '@testing-library/react-native';
import { AddRoom } from '../../components/adminView/AddRoomView';
import { Room } from '../../domain/RoomType';



const testRoom: Room = {
    id: 'A4021',
    patientId: "",
    lastUpdated: Date.now().toString(),
    heartRate: [],
    respirationRate: [],
    oxygenLevel: [],
    notes: []
}

test("renders correctly", () => {
    const tree = render(<AddRoom  />).toJSON();
    expect(tree).toMatchSnapshot();
});
