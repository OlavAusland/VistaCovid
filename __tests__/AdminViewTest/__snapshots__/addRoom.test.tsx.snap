// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "powderblue",
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "flex": 1,
            "padding": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#0274A1",
              "fontSize": 30,
              "letterSpacing": 5,
            }
          }
        >
          Add Room
        </Text>
      </View>
      <View>
        <TextInput
          onChangeText={[Function]}
          placeholder="room number"
          style={
            Object {
              "borderWidth": 1,
              "height": 40,
              "margin": 12,
              "padding": 10,
            }
          }
          testID="roomNumber"
        />
        <View
          style={
            Object {
              "backgroundColor": "powderblue",
              "padding": 20,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Add Room
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
